drupal/modules/lagoon/lagoon_search/lagoon_search.install

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/**
+ * @file
+ * Update hooks for lagoon search.
+ */
+
+/**
+ * Ensures that the solr configuration is up-to-date.
+ *
+ * The latest version of solr changes the path and core, we need to ensure that
+ * all sites that are using Lagoon Search will have this updated. The install
+ * configuration reflects the correct configuration.
+ *
+ * @see https://github.com/govCMS/govcms8lagoon/pull/143
+ */
+function lagoon_search_update_8101() {
+  $config_factory = \Drupal::configFactory();
+  $config = $config_factory->getEditable('search_api.server.lagoon_solr');
+
+  if (!$backend_config = $config->get('backend_config')) {
+    // This means that the Lagoon Solr configuration was not found so
+    // we have nothing to update.
+    \Drupal::logger('lagoon_search')->notice('Skipping lagoon_search_update_8101, nothing to update');
+    return;
+  }
+
+  // Update the configuration to be compatible with the new Solr version.
+  $backend_config['connector_config']['path'] = '/';
+  $backend_config['connector_config']['core'] = 'drupal';
+
+  $config->set('backend_config', $backend_config);
+  $config->save(TRUE);
+
+  $new_config = $config_factory->getEditable('search_api.server.lagoon_solr')->get('backend_config');
+
+  if ($new_config['connector_config']['path'] != '/') {
+    \Drupal::logger('lagoon_search')->notice('Backend path configuration was not updated.');
```

Comment:
do we want to throw this as an updater exception? if not - can you please add a comment to the codebase why we are not throwing that exception. 

Usually, you would throw an exception if the purpose of the update was not fulfilled.

Reply:
We didn't want to throw an exception as this would hinder the deployment process, rather we would monitor the notice and react accordingly for sites that were not able to apply the update.